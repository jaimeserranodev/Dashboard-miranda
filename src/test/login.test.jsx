import React from 'react';
import { render, screen } from '@testing-library/react';
import Button from '../styles/variables';
import '@testing-library/jest-dom/extend-expect';



//------------------TEST PARA BUTTON------------------------//

describe('Button', () => {

  it('Debería tener el texto correcto', () => {
    render(<Button data-cy="submit">Login</Button>);
    const button = screen.getByText('Login');

    expect(button).toBeInTheDocument();
  });

  it('Debería tener ese color de bacñground', () => {
    const { container } = render(<Button data-cy="submit">Login</Button>);
    
    expect(container.firstChild).toMatchSnapshot("background-color: #4EB99F;");
  });

  it('Deberia tener el color del prop', () => {
    const { container } = render(<Button data-cy="submit" color="#FF0000">Login</Button>);
    
    expect(container.firstChild).toMatchSnapshot('background-color: #FF0000');
  });

  it('Deberia tener los estilos de las props de Button', () => {
    render(
      <Button
        data-cy="submit"
        style={{ borderRadius: '10px', fontSize: '24px' }}
      >
        Login
      </Button>
    );
    const button = screen.getByRole('button', { name: 'Login' });

    expect(button).toHaveStyle('border-radius: 10px');
    expect(button).toHaveStyle('font-size: 24px');
  });

  
  
});