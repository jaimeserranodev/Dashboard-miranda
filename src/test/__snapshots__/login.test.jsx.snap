// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Deberia tener el color del prop: background-color: #FF0000 1`] = `
<button
  class="sc-jzuGiw doIclZ"
  color="#FF0000"
  data-cy="submit"
>
  Login
</button>
`;

exports[`Button Debería tener ese color de bacñground: background-color: #4EB99F; 1`] = `
<button
  class="sc-jzuGiw doIclZ"
  data-cy="submit"
>
  Login
</button>
`;

exports[`Button should have the correct background color 1`] = `
<button
  class="sc-jzuGiw doIclZ"
  data-cy="submit"
>
  Login
</button>
`;

exports[`Button should have the correct background color: background-color: #4EB99F; 1`] = `
<button
  class="sc-jzuGiw doIclZ"
  data-cy="submit"
>
  Login
</button>
`;

exports[`Button should have the correct background color: background-color: #FF0000 1`] = `
<button
  class="sc-jzuGiw doIclZ"
  color="#FF0000"
  data-cy="submit"
>
  Login
</button>
`;
